### PR TITLE
Action Item List

### PR DESCRIPTION
**Pull request by thudsonbu**
## Description

This pr adds a view for user's action items. User's can currently only filter by completed and uncompleted but it is quite a bit better then going to go look around in movies. One big benefit is that the drawer now has a purpose. Heres a quick demo:

https://user-images.githubusercontent.com/54583311/204010150-e4daaf37-2cbc-4442-a007-975518bd47cc.mov

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/lib/controllers/user.js [hunk 1]
```diff
@@ -159,4 +160,58 @@ module.exports = {
...
+
+    if (completed) {
+      filters.push({ $match: { completed: JSON.parse(completed) } });
+    }
+
+    if (!skip) {
+      skip = 0;
+    } else {
+      skip = parseInt(skip);
+    }
```
Comment (zbarnz) [inline api/lib/controllers/user.js:178]:
```suggestion
skip ? parseInt(skip) : 0
```

Just curious if you think this is more readable. Im always torn between doing it the two ways

File: api/lib/controllers/user.js [hunk 2]
```diff
@@ -159,4 +160,58 @@ module.exports = {
...
+      },
+      {
+        $project: {
+          action_items: {
+            $slice: ["$action_items", skip, limit],
+          },
+          count: { $size: "$action_items" },
+        },
+      },
+    ]);
```
Comment (zbarnz) [inline api/lib/controllers/user.js:213]:
```
 _ __ ___   ___  _ __   __ _  ___  
| '_ ` _ \ / _ \| '_ \ / _` |/ _ \ 
| | | | | | (_) | | | | (_| | (_) |
|_| |_| |_|\___/|_| |_|\__, |\___/ 
                        __/ |      
                       |___/       
```

File: www/components/Layout/Drawer/Drawer.js [hunk 3]
```diff
@@ -25,26 +24,19 @@ const Drawer = () => {
...
-          startIcon={<DescriptionIcon />}
+          startIcon={<TaskAltIcon />}
           variant="text"
           size="large"
-          className={styles.icon_button + " " + styles.purple}
+          className={styles.icon_button + " " + styles.blue}
+          onClick={() => router.push(`/user/${user._id}/tasks`)}
         >
-          Drafts
+          Action items
```
Comment (zbarnz) [inline www/components/Layout/Drawer/Drawer.js:39]: No way, the drawer actually works now!?!?!!?!

**Comment by zbarnz** [comment]
I like it, the only thing I would say from a UX perspective is the checkboxes to mark complete are a little confusing. I think complete should be a different indicator because checkboxes on a table usually indicate multiselect like if a user wanted to delete a bunch of action items.

**Review by zbarnz — APPROVED** [review]

File: api/lib/controllers/user.js (hunk 1 above)
Comment (thudsonbu) [reply api/lib/controllers/user.js:178]: I find it easier to read with the separate blocks but I don't really think it matters either way unless the ternary takes up multiple lines or is nested in some way.

File: www/components/Layout/Drawer/Drawer.js (hunk 3 above)
Comment (thudsonbu) [reply www/components/Layout/Drawer/Drawer.js:39]: It actually has a function!

**Comment by thudsonbu** [comment]
@zbarnz Definitely a valid point about the table checkboxes. I can't think of a better way to do it though so I think we should leave it for now.

**Pull request merged**